lib/main.js

```diff
@@ -221,3 +219,9 @@ const main = async () => {
 }
 
 main()
+  .catch(async (err) => {
+    console.error(err)
+    process.exitCode = 1
```

Comment:
For this particular case, I think we can just do `process.exit(1)` below `.off()`, as `.off()` will make sure QEMU is dead, right?

I'm thinking that here we set `exitCode` to 1, but then while `.off()` executes, something puts 0 back on it, and then we incorrectly exit successfully

Reply:
Nothing puts 0 back on it. We can run process.exit, but I simply wanted to avoid that call.

Reply:
The child event handlers do set `exitCode`, potentially to zero because of `code || 0`. What happens if we set it to 1 here, and then the child receives SIGINT or SIGTERM while `.off()` is still running?

Reply:
That code is no longer a thing.

Reply:
I still see it here, so maybe you forgot to push? :P

Reply:
Some of it is still there, but it cannot set the exitCode to 0. Which means it is fine, right?

Reply:
Maybe worth doing `process.exit(1)` explicitly, just in case we forget about it and accidentally introduce a weird bug in the future?

Reply:
process.exit is really weird as some of the internal  teardown code will not happen.